script/post-install.ts

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env ts-node
+
+import * as Path from 'path'
+import { spawnSync, SpawnSyncOptions } from 'child_process'
+
+import * as glob from 'glob'
+
+const root = Path.dirname(__dirname)
+
+const options: SpawnSyncOptions = {
+  cwd: root,
+  stdio: 'inherit',
+}
+
+function findYarnVersion(callback: (path: string) => void) {
+  glob('vendor/yarn-*.js', (error, files) => {
+    if (error != null) {
+      throw error
+    }
+
+    // this ensures the paths returned by glob are sorted alphabetically
+    files.sort()
+
+    // use the latest version here if multiple are found
+    const latestVersion = files[files.length - 1]
```

Comment:
If `files` is being sorted lexicographically, there is a chance `latestVersion` won't actually be the latest version.

Reply:
I'm not quite sure how these are sorted, just that `glob`  mentions that they are sorted by default.

Given it's a manual process to change the vendored version of yarn, I'm not sure how defensive we need to be in here to handle multiple yarn packages being present on disk. We can guard against this by using the `engines` property in `package.json`, to ensure a minimum version is used.

What's your gut feeling on this?

Reply:
I say we leave it since we have to update yarn manually.